lib/valueflow.cpp

```diff
@@ -4489,7 +4490,7 @@ static void valueFlowLifetimeFunction(Token *tok, TokenList &tokenlist, ErrorLog
                 tok->next(), tokenlist, errorLogger, settings);
         }
     } else if (memtok && Token::Match(tok->astParent(), ". push_back|push_front|insert|push|assign") &&
-               astIsContainer(memtok)) {
+               (container = astIsContainer(memtok)) && !container->stdStringLike) {
```

Comment:
I think it would be better to have a function `astIsStdStringLike`(or maybe `astIsContainerOfFunadamentalType` would be more appropriate here) and then you can just do `astIsContainer(memtok) && !astIsStdStringLike(memtok)`. I would prefer that than trying to create a variable here which kind of clunky.